From:	Matthew Garrett <mjg@redhat.com>
Date:	Thu, 20 Sep 2012 10:41:01 -0400
Subject: [PATCH V2 06/10] Restrict /dev/mem and /dev/kmem in secure boot setups
Patch-mainline: Not yet, reviewing
References: fate#314486
Target: SLE-11 SP3

Allowing users to write to address space makes it possible for the kernel
to be subverted. Restrict this when we need to protect the kernel.

Signed-off-by: Matthew Garrett <mjg@redhat.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
Automatically created from "patches.suse/0006_Restrict__dev_mem_and__dev_kmem_in_secure_boot_setups_v2.patch" by xen-port-patches.py

--- sle11sp3.orig/drivers/xen/char/mem.c	2012-10-19 14:06:42.000000000 +0200
+++ sle11sp3/drivers/xen/char/mem.c	2012-12-04 13:55:56.000000000 +0100
@@ -120,6 +120,9 @@ static ssize_t write_mem(struct file *fi
 	if (p != *ppos)
 		return -EFBIG;
 
+	if (!capable(CAP_COMPROMISE_KERNEL))
+		return -EPERM;
+
 	while (count > 0) {
 		sz = size_inside_page(p, count);
 
